src/parsers/json/Parser.h

```diff
@@ -39,16 +38,16 @@ class Parser : public SEMBA::Parsers::Parser {
 
     Parser(const std::string& filename);
     Data read() const;
-    DataExtended readExtended() const;
+    ProblemDescription readExtended() const;
     
 private:
-    
-    json readSolverOptions(const json&) const;
+    json readSolverOptions(const json&, const std::string& key = "solverOptions") const;
     PMGroup readPhysicalModels(const json&) const;
-    std::unique_ptr<Geometry::Mesh::Geometric> readGeometricMesh(const PhysicalModel::Group<>&, const json&) const;
-	void readConnectorOnPoint(PMGroup& pMG, Geometry::Mesh::Geometric& mesh,  const json&) const;
-    Source::Group<>* readSources(Geometry::Mesh::Geometric& mesh, const json&) const;
-    OutputRequest::Group<>* readOutputRequests(Geometry::Mesh::Geometric& mesh, const json&) const;
+    PMGroup readExtendedPhysicalModels(const json& j) const;
+    std::unique_ptr<Geometry::Mesh::Unstructured> readUnstructuredMesh(const PhysicalModel::Group<>&, const json&) const;
```

Comment:
This could be read directly as a type Mesh::Unstructured. I think there is no need to force the unique_ptr here. If the user wants a unique_ptr for some reason, he can use that object to create it. Maybe we could discuss it because other "reads" are using the same philosophy and this would be a significant change, not really prioritary...